lldpd: fixes

systemcrash opened the pull request:
Supplementary fix for PR #14193

Tested on 22.03.5

without this fix, the `-k` param is never applied via setting `lldp_no_version`

cc @howels @stintel

systemcrash committed:
lldpd: fix -k 'lldp_no_version' row

Supplementary fix for PR #14193 and commit
b67182008fd124706be0ec3ce67347447554ffd5

Tested on 22.03.5

Signed-off-by: Paul Donald <newtwen@gmail.com>

systemcrash committed:
lldpd: spell fixes

Supplementary fix for PR #14193

Tested on 22.03.5

Signed-off-by: Paul Donald <newtwen@gmail.com>

systemcrash commented:
@stintel if the opportunity arises, `lldpd` was bumped to [1.0.18](https://github.com/lldpd/lldpd/releases/tag/1.0.18) last month. 



Can at least the 1.0.17 version bump also get cherry picked to `origin/openwrt-22.03` branch please (as well as `origin/openwrt-23.05`)?

howels commented:
> Supplementary fix for PR #14193
> 
> Tested on 22.03.5
> 
> without this fix, the `-k` param is never applied via setting `lldp_no_version`
> 
> cc @howels @stintel

Thanks for catching these, sorry that they slipped through.

systemcrash commented:
No worries, you can become home blind after working with something for
awhile 😵‍💫

systemcrash committed:
lldpd: fix a paste error
from commit 1be2088a5247b2cfabe8be991c1e52ddaf780a16

The original PR #13018 did not exhibit this.

Tested on 22.03.5

Signed-off-by: Paul Donald <newtwen@gmail.com>

systemcrash committed:
lldpd: remove unneeded quotes and variable quoting

from commit 3ce909914a12647bec52bcee0a162dd6d158a4f6

'capabilities enabled x' where x is a string of CSV

Tested on 22.03.5

Signed-off-by: Paul Donald <newtwen@gmail.com>

systemcrash committed:
lldpd: remove unneeded quotes

from commit 3ce909914a12647bec52bcee0a162dd6d158a4f6

Tested on 22.03.5

Signed-off-by: Paul Donald <newtwen@gmail.com>

systemcrash committed:
lldpd: remove unneeded quotes and variable quoting

from commit 24176a6bdd8f26040a97960868fd0d9ee968d695

Tested on 22.03.5

Signed-off-by: Paul Donald <newtwen@gmail.com>

systemcrash committed:
lldpd: remove unneeded quotes and variable quoting

from commit c98ee4dbb3db0f064d990941cdd82e872da76946

agent-type takes 1 of 3 possible keywords which do not require quoting:

         configure lldp agent-type nearest-bridge | nearest-non-tpmr-bridge
         | nearest-customer-bridge

Tested on 22.03.5

Signed-off-by: Paul Donald <newtwen@gmail.com>

systemcrash committed:
lldpd: remove unneeded quotes and variable quoting

from commit ac771313ebedd2c4bfda8adef47650d45d77c32d

portidsubtype takes 1 of 2 possible keywords which do not need quoting:

         configure lldp portidsubtype ifname | macaddress

The third keyword 'local' is used in the syntax when individual ports
are being defined:

         configure [ports ethX [,…]] lldp portidsubtype local value

When this syntax is used, quoting is useful (see test cases for lldpd).
In the init file, the 'local' syntax is unused.

Tested on 22.03.5

Signed-off-by: Paul Donald <newtwen@gmail.com>

systemcrash committed:
lldpd: remove unneeded quotes

from commit a5f715da713304972467612d6934130ce3aa2837

Tested on 22.03.5

Signed-off-by: Paul Donald <newtwen@gmail.com>

systemcrash committed:
lldpd: refactor out ifaces derivation; reuse function

from commit 909f0630663d7ae1471c76154516e3299992a150

Now pass two params to get_config_cid_ifaces() for:

cid_interface
interface

Each of which is a CSV of interfaces.

Tested on 22.03.5

Signed-off-by: Paul Donald <newtwen@gmail.com>

systemcrash committed:
lldpd: fix error "sh: XXXms: bad number"

from commit 3ce909914a12647bec52bcee0a162dd6d158a4f6

The lldpd man page says that "configure lldp tx-interval" can
specify an interval value in milliseconds by appending a "ms" suffix to
the figure. Thus mandating string handling, and not integer comparison.

Tested on 22.03.5

Signed-off-by: Paul Donald <newtwen@gmail.com>

systemcrash commented:
> > Supplementary fix for PR #14193
> > Tested on 22.03.5
> > without this fix, the `-k` param is never applied via setting `lldp_no_version`
> > cc @howels @stintel
> 
> Thanks for catching these, sorry that they slipped through.

If you think the changes look OK, can you approve them? Files Changed -> Review -> Approve

howels reviewed (approved):
Tested and appears to work for the functionality enabled on my system.

Spelling mistakes and unneeded quotes removed. Looks good.

systemcrash commented:
Two weeks and these errors are still left in master...

howels commented:
> Two weeks and these errors are still left in master...

Do we need a label or something to get this reviewed?

huangya90 commented:
>Tested on 22.03.5
Are you sure? There are conflicts if these patches are applied on branch 22.03.

howels commented:
> > Tested on 22.03.5
> 
> Are you sure? There are conflicts if these patches are applied on branch 22.03.

I noticed that comment. This patch should be applied to master.

systemcrash commented:
> > Tested on 22.03.5
> 
> Are you sure? There are conflicts if these patches are applied on branch 22.03.

These patches might not directly apply to 22, but the net resulting code (e.g. from master) *has* at least been tested on 22.

systemcrash commented:
Good that you point this out. These patches should land in master, and can be cherry picked with modifications from there.

howels commented:
> Good that you point this out. These patches should land in master, and can be cherry picked with modifications from there.

Suggest amending the commit comments to reflect that they should be applied to master.

systemcrash commented:
Why? The PR target is the master branch.

The tested tag just points out the lowest compatible version.

howels commented:
> Why? The PR target is the master branch. The tested tag just points out the lowest compatible version.

This is a fix PR for the master branch and requires a prior commit from master to apply on the tested tag branch.

Maybe it's fine, but seems like a reviewer was missing the previous commits.

robimarko commented:
Thanks! Rebased on top of main and merged!

systemcrash commented:
Sure - you're welcome!
